src/rules/preferConditionalExpressionRule.ts

```diff
@@ -107,3 +117,20 @@ function getAssigned(node: ts.Statement, sourceFile: ts.SourceFile): ts.Expressi
 function nodeEquals(a: ts.Node, b: ts.Node, sourceFile: ts.SourceFile): boolean {
     return a.getText(sourceFile) === b.getText(sourceFile);
 }
+
+function detectReturn(ifStatement: ts.IfStatement): boolean {
```

Comment:
instead of duplicating the `detect` function, I'd suggest modifying `getAssigned`:
* add a condition for ReturnStatement
* return an object to indicate what kind of statement it is (return or assignment) including the expression if it is an assignment
* `nodeEquals` checks if either both are returns OR both are assignments and the expressions are equal

Otherwise `detectReturn` and `hasReturn` would also need to handle the `checkElseIf` option